hanish/zipcode.py

```diff
@@ -0,0 +1,168 @@
+# hanish.zipcode
+# Utility for mapping zip codes to latitude, longitude coords.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Mon May 08 12:11:01 2017 -0400
+#
+# Copyright (C) 2016 Bengfort.com
+# For license information, see LICENSE.txt
+#
+# ID: zipcode.py [] benjamin@bengfort.com $
+
+"""
+Utility for mapping zip codes to latitude, longitude coords. Loads an
+in-memory sqlite database for quickly querying specific zip codes.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import csv
+import sqlite3
+
+from .exceptions import DatabaseError, HanishValueError
+
+
+SCHEMA = (
+    "CREATE TABLE IF NOT EXISTS zipcodes ("
+        "zipcode TEXT, "
+        "latitude REAL, "
+        "longitude REAL"
+    ")"
+)
+
+
+##########################################################################
+## ZipCodeDB
+##########################################################################
+
+class ZipCodeDB(object):
```

Comment:
Thanks! I viewed this as my "one extension" contribution. Primarily I wanted to show the use of in-memory databases as a counterpart to the `ExpiringDict` cache in the Dark Sky API.; they are also very performant. 

There is also a clear extension to relational data, e.g. the mapping of city names to zip codes to support queries like `weather in Springfield` or `weather in New York`. In the first case, disambiguation would be required, which is easier with a database query than dictionaries. We could also guess "the most likely" by also keep track in another table the frequency of queries. My thought was that it would be easier to iterate on this data structure for future development than simpler in-memory data types.